src/main/java/snipe/storage/Storage.java

```diff
@@ -0,0 +1,129 @@
+package snipe.storage;
+
+import snipe.core.TaskList;
+import snipe.exception.SnipeException;
+import snipe.task.Deadline;
+import snipe.task.Event;
+import snipe.task.Task;
+import snipe.task.ToDo;
+
+import java.util.ArrayList;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.Files;
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.IOException;
+
+/**
+ * The {@code Storage} class handles the reading and writing of a user's task list to and from a file.
+ * It is responsible for loading tasks from the file when the application starts and saving tasks back to the file when the application exits or tasks are modified.
+ */
+public class Storage {
+    private final Path filePath;
+
+    /**
+     * Constructs a {@code Storage} object with the specified file path.
+     *
+     * @param filePath The path to the file where the task list is stored.
+     */
+    public Storage(String filePath) {
+        this.filePath = Paths.get(filePath);
+    }
+
+    /**
+     * Ensures the directory and file exist, creating them if necessary.
+     *
+     * @throws IOException If an input or output error occurs while creating the directory or file.
+     */
+    public void initialise() throws IOException {
```

Comment:
Perhaps the name of this method could be more specific, e.g. initialiseDatabase?